ext/har/logger_test.go

```diff
@@ -0,0 +1,268 @@
+
+    entries := logger.GetEntries()
+    require.Len(t, entries, 1, "Should have one log entry")
+    entry := entries[0]
+
+    // Convert headers to maps for easier checking
+    reqHeaders := make(map[string]string)
+    for _, h := range entry.Request.Headers {
+        reqHeaders[h.Name] = h.Value
+    }
+    assert.Equal(t, "custom-value", reqHeaders["X-Custom-Header"], "Request header value should match")
+
+    respHeaders := make(map[string]string)
+    for _, h := range entry.Response.Headers {
+        respHeaders[h.Name] = h.Value
+    }
+    assert.Equal(t, "test-value", respHeaders["X-Test-Header"], "Response header value should match")
+}
+
+func TestHarLoggerSaveAndClear(t *testing.T) {
+    var exportedEntries []Entry
+    exportFunc := func(entries []Entry) {
+        exportedEntries = append(exportedEntries, entries...)
+    }
+    logger := NewLogger(exportFunc)
+    defer logger.Stop()
+
+    background := httptest.NewServer(ConstantHandler("test"))
+    defer background.Close()
+
+    proxyServer := createTestProxy(logger)
+    defer proxyServer.Close()
+
+    client := createProxyClient(proxyServer.URL)
+
+    resp, err := client.Get(background.URL)
+    require.NoError(t, err, "Should send request")
+    resp.Body.Close()
+
+    time.Sleep(200 * time.Millisecond)
+
+    entries := logger.GetEntries()
+    require.Len(t, entries, 1, "Should have one log entry")
+
+    // Save to file
+    tmpDir := t.TempDir()
+    harFilePath := filepath.Join(tmpDir, "test.har")
+    err = logger.SaveToFile(harFilePath)
+    require.NoError(t, err, "Should save HAR file")
+
+    // Verify file contents
+    harData, err := os.ReadFile(harFilePath)
+    require.NoError(t, err, "Should read HAR file")
+
+    var har Har
+    err = json.Unmarshal(harData, &har)
+    require.NoError(t, err, "Should parse HAR JSON")
+    assert.Len(t, har.Log.Entries, 1, "Saved HAR should have one entry")
+    assert.Equal(t, "1.2", har.Log.Version, "HAR version should be 1.2")
+
+    // Clear logger
+    logger.Clear()
+    entries = logger.GetEntries()
+    assert.Empty(t, entries, "Should have no entries after clear")
+}
+
+func TestHarLoggerExportInterval(t *testing.T) {
+    var exportedEntries []Entry
+    exportFunc := func(entries []Entry) {
+        exportedEntries = append(exportedEntries, entries...)
+    }
+    logger := NewLogger(exportFunc, WithExportInterval(500*time.Millisecond))
+    defer logger.Stop()
+
+    background := httptest.NewServer(ConstantHandler("test"))
+    defer background.Close()
+
+    proxyServer := createTestProxy(logger)
+    defer proxyServer.Close()
+
+    client := createProxyClient(proxyServer.URL)
+
+    // Send 3 requests
+    for i := 0; i < 3; i++ {
+        resp, err := client.Get(background.URL)
+        require.NoError(t, err, "Should send request")
+        resp.Body.Close()
+        time.Sleep(200 * time.Millisecond)
+    }
+
+    // Wait for export interval
+    time.Sleep(600 * time.Millisecond)
```

Comment:
Remove time.Sleep from all your tests and use testify directly inside the callback handler to have a determinisitc behavior (maybe using request counter instead of time based calllback call).
And maybe use a sync.WaitGroup to wait for the finish of the callback, to make sure it's executed.